commit 4aa6d46d1711758ee18d078b76175b807ac55438
Author: Baruch Siach <baruch@tkos.co.il>
Date:   Thu Nov 29 12:40:11 2018 +0200

    net: phy: sfp: correct store of detected link modes
    
    [ Upstream commit d7f7e0018b96fd1a30a968faa9464eb57372c1ec ]
    
    The link modes that sfp_parse_support() detects are stored in the
    'modes' bitmap. There is no reason to make an exception for 1000Base-PX
    or 1000Base-BX10.
    
    Fixes: 03145864bd0f ("sfp: support 1G BiDi (eg, FiberStore SFP-GE-BX) modules")
    Signed-off-by: Baruch Siach <baruch@tkos.co.il>
    Acked-by: Russell King <rmk+kernel@armlinux.org.uk>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/phy/sfp-bus.c b/drivers/net/phy/sfp-bus.c
index 83060fb349f4..ad9db652874d 100644
--- a/drivers/net/phy/sfp-bus.c
+++ b/drivers/net/phy/sfp-bus.c
@@ -162,7 +162,7 @@ void sfp_parse_support(struct sfp_bus *bus, const struct sfp_eeprom_id *id,
 	/* 1000Base-PX or 1000Base-BX10 */
 	if ((id->base.e_base_px || id->base.e_base_bx10) &&
 	    br_min <= 1300 && br_max >= 1200)
-		phylink_set(support, 1000baseX_Full);
+		phylink_set(modes, 1000baseX_Full);
 
 	/* For active or passive cables, select the link modes
 	 * based on the bit rates and the cable compliance bytes.
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1017_4aa6d46d1711758ee18d078b76175b807ac55438.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1017_4aa6d46d1711758ee18d078b76175b807ac55438.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
